reefperf/cloud_driver.py

```diff
@@ -0,0 +1,55 @@
+from abc import ABCMeta, abstractmethod
+from functools import lru_cache
+
+import libcloud
+
+from reefperf.cloud_node import LCCloudScaleNode
+from reefperf.ssh_key_generator import ParamikoRSAKeyGenerator
+
+
+class CloudDriver(object):
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def create_node(self, cloud_node_cfg):
+        pass
+
+
+# LC is a short for apache libcloud. It means that
+# all classes with prefix LC using this libcloud for operating on cloud nodes
+
+
+class LCCloudScaleDriver(CloudDriver):
+    SSH_KEY_LENGTH = 4096
+
+    def __init__(self, credentials):
+        driver_cls = libcloud.get_driver(
+            libcloud.DriverType.COMPUTE,
+            libcloud.DriverType.COMPUTE.CLOUDSCALE,
+        )
+        self._driver = driver_cls(credentials['api_token'])
+
+    @property
+    @lru_cache(maxsize=1)
+    def images(self):
+        return {image.id: image for image in self._driver.list_images()}
+
+    @property
+    @lru_cache(maxsize=1)
+    def sizes(self):
+        return {size.id: size for size in self._driver.list_sizes()}
+
+    def create_node(self, cloud_node_cfg):
```

Comment:
`def create_node(self, node_name, size, image, connection):`

Reply:
There is no doubt, that current version is less readable than yours. But, method `_create_app_nodes` in `Theater` class have to have a possibility for invoking `create_node` method from all existing `CloudDriver` subclasses. So, single container parameter is necessary here. Maybe raw python dict is improper here. Maybe we should create some DriverConfig class for that responsibility?

Reply:
Optional introduction of DriverConfig classes I want add in separate PR

Reply:
I think `create_node` should just have a different set of arguments for each `CloudDriver`, the abstract version should have a common set of arguments. `DriverConfig` doesn't really solve any problem, it's just another version of `clould_node_cfg` dict but in a slightly different form, still makes any create_node implementation incompatible with arguments not meant for it... my version just makes it explicit